File: 240424/정수 명령 처리 6/process-numeric-commands-6.py
import heapq

class PriorityQueue:
    def __init__(self):          
        self.items = []
                
    def push(self, item):        
        heapq.heappush(self.items, -item)
                
    def empty(self):             
        return not self.items
                
    def size(self):              
        return len(self.items)
        
    def pop(self):
        if self.empty():
            raise Exception("PriorityQueue is empty")
            
        return -heapq.heappop(self.items)
                
    def top(self):
        if self.empty():
            raise Exception("PriorityQueue is empty")                  
        return -self.items[0]

pq = PriorityQueue()

n = int(input())
for _ in range(n):
    cmd = input()
    if cmd.startswith("push"):
        x = int(cmd.split()[1])
        pq.push(x)
    elif cmd == "pop":
        print(pq.pop())
    elif cmd == "size":
        print(pq.size())
    elif cmd == "empty":
        print(1 if pq.empty() else 0)
    else:
        print(pq.top())

File: 240424/정수 명령 처리 6/process-numeric-commands-6.cpp
#include <iostream>
#include <queue>
#include <string>

using namespace std;

int main() {
    priority_queue<int> pq;

    int n, t;
    cin >> n;
    
    string cmd;
    for (int i = 0; i < n; i++) {
        cin >> cmd;
        if (cmd == "push") {
            cin >> t;
            pq.push(t);
        } else if (cmd == "size") {
            cout << pq.size() << endl;
        } else if (cmd == "pop") {
            cout << pq.top() << endl;
            pq.pop();
        } else if (cmd == "empty") {
            cout << pq.empty() << endl;
        } else if (cmd == "top") {
            cout << pq.top() << endl;
        }
    }

    return 0;
}